src/components/search/detailed/SearchResultsTable.js

```diff
@@ -0,0 +1,168 @@
+/**
+ *
+ * A generic tabular view to display search results.
+ *
+ * @author sriram
+ *
+ */
+
+import React from "react";
+import { useTable, useSortBy } from "react-table";
+import { useTranslation } from "i18n";
+
+import PageWrapper from "components/layout/PageWrapper";
+import TableLoading from "../../utils/TableLoading";
+import ids from "../ids";
+
+import { build } from "@cyverse-de/ui-lib";
+
+import {
+    Button,
+    CircularProgress,
+    Paper,
+    useMediaQuery,
+    useTheme,
+    Table,
+    TableBody,
+    TableCell,
+    TableContainer,
+    TableSortLabel,
+    TableHead,
+    TableRow,
+    Toolbar,
+} from "@material-ui/core";
+
+const SearchResultsTable = ({
+    baseId,
+    columns,
+    data,
+    loading,
+    fetchMore,
+    canFetchMore,
+    isFetchingMore,
+    initialSortBy,
+    onSort,
+}) => {
+    const {
+        toggleSortBy,
+        getTableProps,
+        headerGroups,
+        rows,
+        prepareRow,
+    } = useTable(
+        {
+            columns,
+            data,
+            manualSortBy: true,
+            disableMultiSort: true,
+            initialState: {
+                sortBy: initialSortBy,
+            },
+        },
+        useSortBy
+    );
+    const theme = useTheme();
+    const isMobile = useMediaQuery(theme.breakpoints.down("xs"));
+    const tableId = build(baseId, ids.TABLE_VIEW);
+    const { t } = useTranslation("search");
+
+    return (
+        <PageWrapper appBarHeight={isMobile ? 210 : 225}>
```

Comment:
Again, not for this PR, but would be good to figure out a way to not have to put hard values here